### PR TITLE
add PillLabels

### LINKED ISSUE #72: Add Pill / Label component

### PR DESCRIPTION
**Pull request by DasProffi**
closes #72
Unbenannt

### CODE REVIEW

**Review by MaxSchaefer — APPROVED** [review]
@DasProffi looks good! If you want, you can provide the implementation for the other representation in the issue as well.

File: app/components/PillLabel.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+const PillLabel = ({ count = 0, state = 'unscheduled' }: PillLabelProps) => {
+  // case 'unscheduled' is default
+  let labelNumber = '1'
+  let text = 'unscheduled'
+  switch (state) {
+    case 'inProgress': labelNumber = '2'; text = 'in progress'; break
+    case 'done': labelNumber = '3'; text = 'done'; break
+    default: break
+  }
```
Comment (use-to) [inline app/components/PillLabel.tsx:16]: use an map data structure instead
`switch` can contain any type of logic -a `map` is more restrictive

File: app/components/PillLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  // TODO Translation
+  return (
+    <div className={tw(`flex flex-row w-2/3 px-2 rounded-md justify-between bg-hw-label-${labelNumber}-background`)}>
+      <div className={tw(`flex flex-row items-center text-hw-label-${labelNumber}-text`)}>
+        <div className={tw(`rounded-full w-2 h-2 bg-hw-label-${labelNumber}-accent`)} />
+        <div className={tw('w-2')} />
+        {text}
```
Comment (use-to) [inline app/components/PillLabel.tsx:23]: Is it the right font? Is `Inter` the default one?
There is the twind class `font-sans` but I'm not sure whether we have to use it. @JannikWibker ?

**Review by use-to — CHANGES_REQUESTED** [review]

File: app/components/PillLabel.tsx (hunk 1 above)
Comment (MaxSchaefer) [reply app/components/PillLabel.tsx:16]: The state can be an enum and should then be used in a map with the displayable text.

File: app/enums/TaskState.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+enum TaskState {
+  unscheduled,
+  inProgress,
+  done
+}
+
+export { TaskState }
```
Comment (use-to) [inline app/enums/TaskState.tsx:7]: Why should we use a enum here?

```suggestion
export const TaskState = {
	unscheduled: {text: "unscheduled", label: 1},
	inProgress: {text: "in progress", label: 2},
	done: {text: "done", label: 3},
}
```

So just use:
```ts
const doneState = TaskState.done;
console.log(doneState.text, doneState.label)
```

**Review by use-to — CHANGES_REQUESTED** [review]

File: app/components/PillLabel.tsx (hunk 2 above)
Comment (use-to) [reply app/components/PillLabel.tsx:25]: @DasProffi just ensure using `Inter` here. We don't have to wait for @JannikWibker with this. It's more a general question of mine.

**Comment by use-to** [comment]
> @DasProffi looks good! If you want, you can provide the implementation for the other representation in the issue as well.

I would skip that for the moment! Just concentrate on other issues/components.
Maybe create a new issues with low priority for that.

File: app/enums/TaskState.tsx (hunk 3 above)
Comment (MaxSchaefer) [reply app/enums/TaskState.tsx:7]: Is also possible.

File: app/components/PillLabel.tsx (hunk 2 above)
Comment (DasProffi) [reply app/components/PillLabel.tsx]: As far as I know, `inter` is the default for tailwind, therefore no additional tag should be required

File: app/components/PillLabel.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { tw } from '@helpwave/common/twind/index'
+import { TaskStates } from '../dataclasses/TaskState'
+
+export type PillLabelProps = {
+  count?: number,
+  state?: {text: string, label: number}
```
Comment (use-to) [inline app/components/PillLabel.tsx:6]: create a type definition for this

File: app/dataclasses/TaskState.tsx [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const TaskStates = {
+  unscheduled: { text: 'unscheduled', label: 1 },
+  inProgress: { text: 'in progress', label: 2 },
+  done: { text: 'done', label: 3 },
```
Comment (use-to) [inline app/dataclasses/TaskState.tsx:4]: Add type definitions

File: app/dataclasses/TaskState.tsx [hunk 6]
```diff
@@ -0,0 +1,5 @@
+export const TaskStates = {
```
Comment (use-to) [inline app/dataclasses/TaskState.tsx:1]:
```suggestion
export const TaskState = {
```

**Review by use-to — CHANGES_REQUESTED** [review]

File: app/dataclasses/TaskState.tsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export const TaskState: { [key in 'unscheduled' | 'inProgress' | 'done']: TaskStateInformation } = {
```
Comment (DasProffi) [inline app/dataclasses/TaskState.tsx:1]: should the key only be a string?

**Review by use-to — APPROVED** [review]

**Review by use-to — APPROVED** [review]

**Pull request merged**